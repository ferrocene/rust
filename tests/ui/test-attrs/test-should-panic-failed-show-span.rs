//@ compile-flags: --test
//@ run-flags: --test-threads=1 --nocapture
//@ run-fail
//@ check-run-results
//@ exec-env:RUST_BACKTRACE=0
//@ normalize-stdout: "finished in \d+\.\d+s" -> "finished in $$TIME"
//@ needs-threads
//@ needs-unwind (panic)

#[test]
#[should_panic]
fn should_panic_with_any_message() {
    panic!("Panic!");
}

#[test]
#[should_panic = "message"]
fn should_panic_with_message() {
    panic!("message");
}

#[test]
#[should_panic]
fn should_panic_with_any_message_does_not_panic() {
    // DON'T PANIC
}

#[test]
#[should_panic = "message"]
fn should_panic_with_message_does_not_panic() {
    // DON'T PANIC
}

#[test]
#[should_panic = "message"]
fn should_panic_with_substring_panics_with_incorrect_string() {
    panic!("ZOMGWTFBBQ");
}
