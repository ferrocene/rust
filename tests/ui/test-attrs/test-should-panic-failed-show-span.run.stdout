
running 5 tests
test should_panic_with_any_message - should panic ... ok
test should_panic_with_any_message_does_not_panic - should panic ... FAILED
test should_panic_with_message - should panic ... ok
test should_panic_with_message_does_not_panic - should panic ... FAILED
test should_panic_with_substring_panics_with_incorrect_string - should panic ... FAILED

failures:

---- should_panic_with_any_message_does_not_panic stdout ----
note: test did not panic as expected at $DIR/test-should-panic-failed-show-span.rs:24:4
---- should_panic_with_message_does_not_panic stdout ----
note: test did not panic as expected at $DIR/test-should-panic-failed-show-span.rs:30:4
---- should_panic_with_substring_panics_with_incorrect_string stdout ----
note: panic did not contain expected string
      panic message: "ZOMGWTFBBQ"
 expected substring: "message"

failures:
    should_panic_with_any_message_does_not_panic
    should_panic_with_message_does_not_panic
    should_panic_with_substring_panics_with_incorrect_string

test result: FAILED. 2 passed; 3 failed; 0 ignored; 0 measured; 0 filtered out; finished in $TIME

